### DISCUSSION TITLE
Clarification on Chemistry Integration in Fine-Covered Areas

### DISCUSSION THREAD

**Discussion by yinmin-liu** [AMReX-Combustion/PeleLMeX#209] [Q&A] [answered]
In the 'AMR extension' section of the 'PeleLMeX Algorithm' documentation, there is a statement that says "PeleLMeX does not need to perform the expensive (and often very under-resolved) chemistry integration in fine-covered areas."

I would appreciate some clarification regarding the location of the chemistry integration in PeleLMeX:

Does the chemistry integration occur on both the coarse level and the fine level, excluding the fine-covered areas?
Alternatively, does the chemistry integration only take place on the coarsest level?
Thank you for your assistance in understanding this aspect of PeleLMeX.

**Comment by esclapez** [comment, marked as answer]
Hi there,

The chemistry integration is performed on the entire AMR hierarchy, from the coarsest to the finest level, excluding fine-covered areas on all but the finest level.
The top-level function integrating the chemistry is [here](https://github.com/AMReX-Combustion/PeleLMeX/blob/f31da9d33e3b56f9c2fd797be090926f47d58dbf/Source/PeleLMReactions.cpp#L9), where you can see the loop on levels. At each level, an `advanceChemistry` or  `advanceChemistryBAChem` function is called where the former uses the standard BoxArray/DistributionMap MultiFabs while the later uses a specific BoxArray/DistributionMap created for the purpose of integrating the chemistry, excluding fine-covered (create [here](https://github.com/AMReX-Combustion/PeleLMeX/blob/f31da9d33e3b56f9c2fd797be090926f47d58dbf/Source/PeleLMUtils.cpp#L636)). Note that this specific BoxArray/DistributionMap can also be created and used on the finest level to deal with load balancing.

I hope this helps !

**Reply by yinmin-liu** [reply]
Thanks for making it clear to me. Appreciate it!
